std/xml.d

```diff
@@ -2045,8 +2045,8 @@ class ElementParser
 
                     immutable(char)* p = startTag.tagString.ptr
                         + startTag.tagString.length;
-                    immutable(char)* q = tag_.tagString.ptr;
-                    text = decode(p[0..(q-p)], DecodeMode.LOOSE);
+                    immutable end = &tag_.tagString[0] - (&startTag.tagString[0] + startTag.tagString.length);
+                    text = decode(p[0 .. end], DecodeMode.LOOSE);
```

Comment:
I propose we split this PR into two parts, one with the uncontroversial changes that can be merged ASAP, and another to rework this ugly bit of code to eliminate the pointer arithmetic / out-of-bounds slicing in a satisfactory way.